kerastuner/applications/augment.py

```diff
@@ -0,0 +1,303 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+import tensorflow.keras as keras
+
+from kerastuner.engine import hypermodel
+
+from tensorflow.keras import layers
+from tensorflow.keras.layers.experimental import preprocessing
+
+# dict of functions that create layers for transforms.
+# Each function takes a factor (0 to 1) for the strength
+# of the transform.
+TRANSFORMS = {
+    'translate_x': lambda x: preprocessing.RandomTranslation(x, 0),
+    'translate_y': lambda y: preprocessing.RandomTranslation(0, y),
+    'rotate': preprocessing.RandomRotation,
+    'contrast': preprocessing.RandomContrast,
+}
+
+
+class HyperImageAugment(hypermodel.HyperModel):
+    """ Builds HyperModel for image augmentation.
+
+    # Arguments:
+        input_shape: Optional shape tuple, e.g. `(256, 256, 3)`.
+        input_tensor: Optional Keras tensor (i.e. output of
+            `layers.Input()`) to use as image input for the model.
+        rotate: A number between [0, 1], a list of two numbers between [0, 1]
+            or None. It configures the range of factor of rotation
+            transform in the augmentation. Default is 0.2.
+        translate_x: A number between [0, 1], a list of two numbers between [0, 1]
+            or None. It configures the range of factor of horizontal translation
+            transform in the augmentation. Default is 0.1.
+        translate_y: A number between [0, 1], a list of two numbers between [0, 1]
+            or None. It configures the range of factor of vertical translation
+            transform in the augmentation. Default is 0.1.
+        contrast: A number between [0, 1], a list of two numbers between [0, 1]
+            or None. It configures the range of factor of contrast
+            transform in the augmentation. Default is 0.1.
+        **kwargs: Additional keyword arguments that apply to all
+            HyperModels. See `kerastuner.HyperModel`.
+
+    # Notes for keyword args that configures range of factor of transform:
+        rotate, translate_x, translate_y and contrast:
+            If two numbers are given, they represent minimum and maximum
+            factors; if one number is given, it represents maximum factor, and
+            minimum factor is 0; if None, the transform is disabled. All factors
+            are within [0, 1].
+    """
+    def __init__(self,
+                 input_shape=None,
+                 input_tensor=None,
+                 rotate=0.2,
+                 translate_x=0.1,
+                 translate_y=0.1,
+                 contrast=0.1,
```

Comment:
Can we add an argument like `max_augmentations=None` to control randomly selecting a few augmentations or running all augs.